mediator/esmFldsExchange_nems_mod.F90

```diff
@@ -753,13 +791,11 @@ subroutine esmFldsExchange_nems(gcomp, phase, rc)
 
     ! to lnd - states and fluxes from atm
     if ( trim(coupling_mode) == 'nems_orig_data') then
-       allocate(flds(21))
+       allocate(flds(16))
```

Comment:
Why is it necessary to have two blocks for defining the fields from atm to land? For example, Sa_u,v when they come from a data component and Sa_ua,va (aliased to fields containing the 'from_phys' string)?

Reply:
@DeniseWorthen I have two block in here because one for coupling with data atmosphere (GSWP3 or ERA5) and other for active atmosphere. This is required since CDEPS uses Sa_u, Sa_v as a convention to define low level wind but Sa_ua, Sa_va is used for active atmosphere. Also, I was checking input/output of CCPP/Physics version of NoahMP with the component version and I found that exiting u and v wind fields provided by the FV3 cap (with Sa_u and Sa_v or Sa_u10 and Sa_v10) is not matching with the one goes to CCPP/Physics version. So, I created new Sa_ua, Sa_va pair to have same input also for the component model. Let me know if you want to change this design. It is not too hard to use Sa_u and Sa_v if we want to be consistent with other components. Probably, we will have a another PR after this to complete the land coupling work and plaining to use restructured NoahMP model for the component (we got no cost extension until next summer). The Sa_pslv is the surface pressure while Sa_prsl is the one for low level pressure. NoahMP uses them to calculate Exner function in both level (See following inputs, https://github.com/ufs-community/ccpp-physics/blob/31a99de05048187b61a2ea8381417b8dbd8db7e0/physics/noahmpdrv.F90#L252). So, I had to provide both of them to NoahMP driver to get similar results with the CCPP/Physics one. BTW, without splitting the physics and enable coupling with the external land model just before surface part, it is hard to get very similar results with component land and we will have always one coupling time step lag in the component land configurations. There is also ongoing work to bring GFDL's LM4 to UFS and that work could also bring capability to split physics to enable call external land component in the same place that CCPP/Phyiscs land is called.

Reply:
@uturuncoglu Thanks. This line concerns me

> I found that exiting u and v wind fields provided by the FV3 cap (with Sa_u and Sa_v or Sa_u10 and Sa_v10) is not matching with the one goes to CCPP/Physics version. 

So the field values exported by the cap are not identical with the values sent to ccpp? That seems worrisome. How different are they ?

Reply:
If I add the two wind fields to CMEPS and use mediator ATM history files, I can see what is exported as ``Sa_u,v`` vs the ``from_phys`` values.

```
    call addfld_from(compatm , 'inst_pres_height_lowest_from_phys')
    call addfld_from(compatm , 'inst_temp_height_lowest_from_phys')
```

I get that ``Sa_u`` and the ``_from_phys`` are identical.

Reply:
@DeniseWorthen thanks for pointing this. Let me double check in my side. Maybe this is changed. If so, I could get rid of those extra variables and update the PR based on it.

Reply:
Thinking about this a little more, I'm guessing that when you coupled land, these fields are no longer identical. Which means that we'd be sending one field for lowest U,V to the ICE and the AO-fluxes and a different field to the land. Is this what we want? Wouldn't that essentially represent a discontinuity at the the land/ocean boundary?

Reply:
@DeniseWorthen I think the fields needs to be consistent across components. BTW, why lowest U,V is used for those components. I general ocean is forced with 10 m u and v like we did with HYCOM. Is there anything special in MOM6 or CICE?

Reply:
CICE does it's own flux calculation from quantities such as bottom u,v,t etc. This is because CICE has thickness categories. 

In UFS, when the ATM calculates the fluxes, we send the taux,tauy to OCN. But as we move towards using the x-grid w/ the aoflux calculation I think we'll want what LND gets to be consistent w/ ICE and what is sent to MED.

Reply:
@uturuncoglu I think we need to resolve the difference here between what we export to ocn/ice or aoflux and what is exported to the land model. Have you had a chance to test this yet?

Reply:
@DeniseWorthen I think You mean using Sa_u and Sa_v rather than using newly introduced coupling fields. Right? I did not test it yet but I could do it quickly once ESMF 8.5 in. If I could get close results with the side-by-side as before then I could clean the unnecessary part and remove those extra coupling fields. Otherwise, it would be nice to keep them as it is. In any case, we have plan to use restructured NoahMP code in the component and get rid of side-by-side RT. That will be the final PR for the land component work.

Reply:
@DeniseWorthen I think so. It was a long time ago. I had to remind to myself. Anyway, I could try to sync the model and try to test it today and update you soon. Sorry it took long.

Reply:
@DeniseWorthen I am still looking in to it. I need to sync the development with the head of ufs-weather-model and then test. I think that old esmf installation is not compatible with recent version of ufs-weather-model and I might need to use my custom build. Anyway, I'll update you about it. It will take little bit longer than I thought but I think I could make it until end of the week.

Reply:
@DeniseWorthen I could sync the fork and test with both Sa_ua/Sa_va and Sa_u/Sa_v. The results seems identical. So, I am plaining to clean the part that introduces *_from_phys variables. I am not sure why it was creating issue in the past but that is great to get rid of those. Thanks for pointing it. I think the updated code will be available today.

Reply:
Oh, that is good news. Thanks.

Reply:
@DeniseWorthen I did more test with the other fields; 

- I replaced Sa_ta with Sa_tbot. I see that Sa_ta vs Sa_tbot has very minor difference. After 1 day of simulation, surface temperature is getting little bit warmer, bias/rmse of latent/sensible heat and surface temperature increases when it is compared with the original run. So, we might want to keep Sa_ta at least for now until get rid of side-by-side run.
- I replaced Sa_qa with Sa_shum. Again, there is a difference between Sa_qa vs Sa_shum but
bias/rmse looks very similar to original run. Again, we could keep this as its sine they are not same fields.
- Lastly, I changed Sa_prsl to Sa_pbot. There are minor differences between those fields. I did not recognize a big difference. Since they are slightly different we could also keep this as its.

Anyway, at this point I am planing to revert only Sa_ua and Sa_ua to Sa_u and Sa_v and update/clean PR like that. Let me know if you have any other idea.

Reply:
@DeniseWorthen I removed Sa_ua and Sa_va from the cooling fields and replace them with Sa_u,Sa_v pair.